### PR TITLE
Feature: add DataStorages

### PR DESCRIPTION
**Pull request by ec**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -38,6 +38,15 @@
     t.index ["user_id"], name: "index_apikeys_on_user_id"
   end
 
+  create_table "data_storages", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8", force: :cascade do |t|
+    t.bigint "user_id", null: false, unsigned: true
+    t.string "title", limit: 64, null: false
+    t.text "data", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["user_id", "title"], name: "index_data_storages_on_user_id_and_title"
```
Comment (calj) [inline db/schema.rb]: this must be a unique index

File: app/api/v2/resource/data_storage.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module API
+  module V2
+    module Resource
+      # data storage C functionality
+      class DataStorage < Grape::API
+        resource :data_storage do
+          desc 'Create an api key',
```
Comment (calj) [inline app/api/v2/resource/data_storage.rb]: Description is wrong here

File: config/barong.yml [hunk 3]
```diff
@@ -16,3 +16,6 @@ document_types:
   - Utility Bill
   - Residental
   - Institutional
+data_storage_titles:
```
Comment (calj) [inline config/barong.yml]: I would set the default value as empty array here to prevent accidentally activating user storage.

Comment (calj) [inline config/barong.yml]: I would name it `user_storage_titles`

File: config/initializers/data_storage_titles.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+# whitelisted data storage titles definitions
+class DataStorageTitles
+  class << self
+    def list
+      @list ||= YAML.load_file(Barong::App.config.barong_config)['data_storage_titles']
```
Comment (calj) [inline config/initializers/data_storage_titles.rb]: default it to [] if unset

Comment (ec) [reply config/initializers/data_storage_titles.rb]: you mean if key `data_storage_titles` dont exist at all?

File: config/barong.yml (hunk 3 above)
Comment (ec) [reply config/barong.yml]: about `user_storage` prefix dont you think its a bit unclear?
user_storage sounds like a storage of users?

Comment (ec) [reply config/barong.yml]: okay. but i will keep possible options as a comment to show possible functionality

Comment (calj) [reply config/barong.yml]: user_personal_storage_titles_whitelist ?

File: config/initializers/data_storage_titles.rb (hunk 4 above)
Comment (calj) [reply config/initializers/data_storage_titles.rb]: yes to not brake on upgrade of barong if we don't update the configuration

**Pull request merged**